### PR TITLE
fix(test): add timestamp to kafka messages in the simulation

### LINKED ISSUE #7774: test: wrong result for `_rw_kafka_timestamp` in the simulation
### Describe the bug

When running `kafka_batch.slt` in the simulation. The test panicked:
```
thread '<unnamed>' panicked at 'query result mismatch:
[SQL] select * from s1 where _rw_kafka_timestamp > '1977-01-01 00:00:00+00:00'
[Diff] (-expected|+actual)
-   1 1
-   2 22
-   3 333
-   4 4444
```

It looks like there is something wrong with the timestamp in the Kafka simulator.

### To Reproduce

```sh
git checkout 075d054c1 # in PR#7733
./risedev sslt -- --kafka-datadir=./scripts/source/test_data ./e2e_test/source/basic/kafka_batch.slt
```

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by wangrunji0408**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR includes the `kafka_batch.slt` simulation test in CI, and adds timestamp to kafka messages to make it pass.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
fix #7774

### CODE REVIEW

File: src/tests/simulation/src/kafka.rs [hunk 1]
```diff
@@ -75,10 +75,19 @@ pub async fn producer(broker_addr: &str, datadir: String) {
             .expect("failed to create topic");
 
         let content = std::fs::read(file.path()).unwrap();
-        // binary message data, a file is a message
-        if topic.ends_with("bin") {
+        let msgs: Box<dyn Iterator<Item = &[u8]> + Send> = if topic.ends_with("bin") {
+            // binary message data, a file is a message
+            Box::new(std::iter::once(content.as_slice()))
+        } else {
+            Box::new(content.split(|&b| b == b'\n'))
```
Comment (BugenZhao) [inline src/tests/simulation/src/kafka.rs:82]:
```suggestion
            Either::Left(std::iter::once(content.as_slice()))
        } else {
            Either::Right(content.split(|&b| b == b'\n'))
```

**Review by BugenZhao — COMMENTED** [review]
LGTM

**Review by BugenZhao — APPROVED** [review]

File: src/tests/simulation/src/kafka.rs (hunk 1 above)
Comment (wangrunji0408) [reply src/tests/simulation/src/kafka.rs:82]: This part is not on a critical path so I thought dynamic dispatch is acceptable. 🤪

Comment (BugenZhao) [reply src/tests/simulation/src/kafka.rs:82]: Just for omitting the `Box<dyn>` annotation. Both LGTM. 🤣

**Pull request merged**